packages/components/src/internal/components/samples/SamplesCreateSubMenu.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { FC, memo, useCallback, useMemo, useState } from 'react';
+import { List } from 'immutable';
+
+import {
+    ALIQUOT_CREATION,
+    App,
+    AppURL,
+    CHILD_SAMPLE_CREATION,
+    DERIVATIVE_CREATION,
+    MenuOption,
+    POOLED_SAMPLE_CREATION,
+    QueryGridModel,
+    QueryModel,
+    SampleCreationType,
+    SampleCreationTypeModal,
+    SubMenu,
+} from '../../..';
+
+interface SamplesCreateSubMenuProps {
+    getOptions: (useOnClick: boolean, disabledMsg: string, itemActionFn: (key: string) => any) => List<MenuOption>;
+    maxParentPerSample: number;
+    isSelectingSamples: (schemaName: string) => boolean;
+    navigate: (url: string | AppURL) => any;
+    menuCurrentChoice?: string;
+    menuText?: string;
+    parentType?: string;
+    parentKey?: string;
+    parentModel?: QueryGridModel;
+    parentQueryModel?: QueryModel;
+    sampleWizardURL?: (targetSampleType?: string, parent?: string) => AppURL;
+    getProductSampleWizardURL?: (targetSampleType?: string, parent?: string, selectionKey?: string) => string | AppURL;
+}
+
+export const SamplesCreateSubMenu: FC<SamplesCreateSubMenuProps> = memo(props => {
```

Comment:
This was factored out of the SM `CreateSamplesSubMenu.tsx` right? Was there a reason for flipping the parts of the name to `SamplesCreateSubMenu` here? Seems a little confusing that in SM, the `CreateSamplesSubMenu` will create a `SamplesCreateSubMenu`. Should this be called something like `BaseCreateSamplesSubMenu` instead.